record_platform_interface/lib/src/types/device.dart

```diff
@@ -0,0 +1,18 @@
+
+class Device {
+  int? id;
+  String? name;
+  String? channel;
+  String? type;
```

Comment:
I think `type` and `isDefault` could be removed.
type; because it should be always audio capture device.
isDefault: because of consistency between platforms and when a user need to select, this info won't help.

Reply:
There are two types of devices.

input devices ( microphone, input sound, etc... ) = ```type { capture }```
output devices (Speakers, Headphones, etc...) = ```type { playback }```

In both cases there is always a default device assigned by the system, it is essential that the user knows which device he has by default.

Reply:
I'm aware of this but this is a package for recording this is not a player.
'isDefault', won't be available for all platforms, this is why I'm wondering if it's wise to populate it if not all can benefit from its value.
`id` should be non nullable.
`name` should be `label` and non nullable also.